onnx/defs/nn/defs.cc

```diff
@@ -2190,6 +2190,198 @@ ONNX_OPERATOR_SET_SCHEMA(
         }
         )ONNX"));
 
+void col2imShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+  // All inputs shapes are required
+  if (!hasNInputShapes(ctx, 3)) {
+    return;
+  }
+
+  // TODO: Assume image_shape has correct spatial dimensions for next validations
+  //       An alternative is get the the number of spatial dimensions as an input
+  if (ctx.getInputType(1)->tensor_type().shape().dim_size() != 1) {
+    fail_shape_inference("image_shape tensor must have rank 1.");
+  }
+  size_t n_input_dims = ctx.getInputType(1)->tensor_type().shape().dim(0).dim_value();
+  std::vector<int64_t> image_shape = {};
+  const TensorProto* image_shape_data = ctx.getInputData(1);
+  if (image_shape_data) {
+    image_shape = ParseData<int64_t>(image_shape_data);
+    if (image_shape.size() != n_input_dims) {
```

Comment:
Suggest replacing these 3 lines with
```cc
unifyDim (n_input_dims, image_shape.size());
```
to handle the case that n_input_dims is unknown or symbolic correctly.

Reply:
done